File: app/src/main/java/com/mpcorporation/snapeffect/Adapter/ScaleCropAdapter.java
package com.mpcorporation.snapeffect.Adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.mpcorporation.snapeffect.Model.ScaleCrop;
import com.mpcorporation.snapeffect.R;

import java.util.List;

public class ScaleCropAdapter extends RecyclerView.Adapter<ScaleCropAdapter.ViewHolder>{
    public interface OnCropClickListener {
        void onCropClick(ScaleCrop option);
    }

    private final List<ScaleCrop> cropOptions;
    private final OnCropClickListener listener;

    public ScaleCropAdapter(List<ScaleCrop> cropOptions, OnCropClickListener listener) {
        this.cropOptions = cropOptions;
        this.listener = listener;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView icon;
        TextView label;

        public ViewHolder(View view) {
            super(view);
            icon = view.findViewById(R.id.filter_icon);
            label = view.findViewById(R.id.filter_name);
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.filter_component, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        ScaleCrop option = cropOptions.get(position);
        holder.label.setText(option.label);
        holder.icon.setImageResource(option.iconResId);
        holder.itemView.setOnClickListener(v -> listener.onCropClick(option));
    }

    @Override
    public int getItemCount() {
        return cropOptions.size();
    }
}

File: app/src/main/java/com/mpcorporation/snapeffect/presentation/common/adapter/ScaleCropAdapter.kt
package com.mpcorporation.snapeffect.presentation.common.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mpcorporation.snapeffect.R
import com.mpcorporation.snapeffect.domain.model.ScaleCrop

class ScaleCropAdapter(
    private val options: List<ScaleCrop>,
    private val onClick: (ScaleCrop) -> Unit
) : RecyclerView.Adapter<ScaleCropAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val icon: ImageView = view.findViewById(R.id.filter_icon)
        val label: TextView = view.findViewById(R.id.filter_name)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.filter_component, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val option = options[position]
        holder.label.text = option.label
        holder.icon.setImageResource(option.iconResId)
        holder.itemView.setOnClickListener { onClick(option) }
    }

    override fun getItemCount(): Int = options.size
}
